pkg/virtualnetwork/mux.go

```diff
@@ -62,12 +63,16 @@ func (n *VirtualNetwork) ServicesMux() *http.ServeMux {
 			return
 		}
 
+		if port < 0 || port > math.MaxUint16 {
```

Comment:
I’d change `port, err := strconv.Atoi(r.URL.Query().Get("port"))` to use `ParseUint` instead, then the check is not needed.